src/dual.jl

```diff
@@ -384,13 +384,35 @@ for pred in UNARY_PREDICATES
     @eval Base.$(pred)(d::Dual) = $(pred)(value(d))
 end
 
-for pred in BINARY_PREDICATES
+# BINARY_PREDICATES = Symbol[:isequal, :isless, :<, :>, :(==), :(!=), :(<=), :(>=)] # prelude.jl
+
+Base.iszero(x::Dual) = iszero(value(x)) && iszero(partials(x))
+
+for pred in [:isequal, :(==)]
+    @eval begin
+        @define_binary_dual_op(
+            Base.$(pred),
+            $(pred)(value(x), value(y)) && $(pred)(partials(x), partials(y)),
+            $(pred)(value(x), y)        && iszero(partials(x)),
+            $(pred)(x, value(y))        && iszero(partials(y)),
+        )
+    end
+end
+
+@define_binary_dual_op(
+    Base.:(!=),
+    (!=)(value(x), value(y)) || (!=)(partials(x), partials(y)),
+    (!=)(value(x), y)        || !iszero(partials(x)),
+    (!=)(x, value(y))        || !iszero(partials(y)),
+)
+
+for pred in [:isless, :<, :>, :(<=), :(>=)]
```

Comment:
Maybe change `BINARY_PREDICATES` to
```julia
const BINARY_PREDICATES = [:isless, :<, :>, :(<=), :(>=)]
```
and use here
```suggestion
for pred in BINARY_PREDICATES
```
? From a quick look at the source code it seems this is the only place where `BINARY_PREDICATES` was used.

Reply:
Seems a confusing name for a subset of them. Could be deleted. I didn't just in case someone has built on it.

Reply:
I don't mind, I just think it's confusing to keep `BINARY_PREDICATES` if it is not used anymore (I assumed you might just have forgotten to remove it). I don't think it would be too bad to reduce the number of functions in `BINARY_PREDICATES` - `UNARY_PREDICATES` etc. does also not include all possible unary functions but only those for which `@define_unary...` can be used.

Reply:
```suggestion
for pred in [:isless, :<, :>, :(<=), :(>=)]  # note that this list, and :(!=), isn't a minimal set, as Base defines some in terms of others. Left intact by #481
```